trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/security/opa/OpaS3SecurityFacadeProvider.java

```diff
@@ -65,7 +66,12 @@ public S3SecurityFacade securityFacadeForRequest(ParsedS3Request request, Option
 
     private SecurityResponse facade(ParsedS3Request parsedS3Request, Optional<String> lowercaseAction, Optional<Identity> identity)
     {
-        OpaRequest opaRequest = opaS3SecurityMapper.toRequest(parsedS3Request, lowercaseAction, opaServerUri, identity);
+        OpaRequestAdapter opaRequestAdapter = opaS3SecurityMapper.toRequest(parsedS3Request, lowercaseAction, opaServerUri, identity);
+
+        if (opaRequestAdapter instanceof OpaRequestAdapter.SecurityResponseWrapper(SecurityResponse securityResponse)) {
+            return securityResponse;
+        }
+        OpaRequest opaRequest = ((OpaRequestAdapter.OpaRequestWrapper) opaRequestAdapter).opaRequest();
```

Comment:
We can split this into two functions, all the stuff below this line could be moved into a function that takes in an `OpaRequest` and returns a `SecurityResponse`.

We can then typeguard with a switch statement without the need to do any casting.

Something like:
```java
    private SecurityResponse sendRequest(OpaRequest opaRequest)
    {
        Map<String, Object> inputDocument = opaS3SecurityMapper.toInputDocument(opaRequest.document());

        Request.Builder builder = preparePost()
                .setUri(opaRequest.opaServerUri())
                .addHeader(CONTENT_TYPE, APPLICATION_JSON_TYPE.getType())
                .setBodyGenerator(jsonBodyGenerator(CODEC, inputDocument));
        opaRequest.additionalHeaders().forEach((name, values) -> values.forEach(value -> builder.addHeader(name, value)));

        Map<String, Object> responseDocument = httpClient.execute(builder.build(), createJsonResponseHandler(CODEC));
        return opaS3SecurityMapper.toSecurityResponse(responseDocument);
    }

    private SecurityResponse facade(ParsedS3Request parsedS3Request, Optional<String> lowercaseAction, Optional<Identity> identity)
    {
        return switch (opaS3SecurityMapper.toRequest(parsedS3Request, lowercaseAction, opaServerUri, identity)) {
            case OpaRequestAdapter.SecurityResponseWrapper(SecurityResponse securityResponse) -> securityResponse;
            case OpaRequestAdapter.OpaRequestWrapper(OpaRequest opaRequest) -> sendRequest(opaRequest);
        };
    }
```